### PR TITLE
created models and static pages

### PR DESCRIPTION
**Pull request by TheTrung**
* Fixed all comments on styles in the first commit . 
* Added Catogries model/controller.
* Added Lesson model / controller.

### CODE REVIEW

File: app/controllers/static_pages_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class StaticPagesController < ApplicationController
+  def home
+  end
+
+  def wordlist
```
Comment (chuanhtuan) [inline app/controllers/static_pages_controller.rb]: remove

File: app/controllers/static_pages_controller.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  end
+
+  def categories
```
Comment (chuanhtuan) [inline app/controllers/static_pages_controller.rb]: remove

File: app/helpers/static_pages_helper.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module StaticPagesHelper
+	def logged_in?
```
Comment (chuanhtuan) [inline app/helpers/static_pages_helper.rb]: indent 2

File: app/models/activity.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Activity < ActiveRecord::Base
+	belongs_to :learnedwords, class_name: "User"
```
Comment (chuanhtuan) [inline app/models/activity.rb]: remove
không dùng thông qua activity

File: app/models/lesson.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class Lesson < ActiveRecord::Base
+	has_many :word
```
Comment (chuanhtuan) [inline app/models/lesson.rb]: indent

File: app/models/question.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class Question < ActiveRecord::Base
+	belongs_to :word
```
Comment (chuanhtuan) [inline app/models/question.rb]: indent

File: app/models/relationship.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class Relationship < ActiveRecord::Base
+	validate :followed_id, presence: true
```
Comment (chuanhtuan) [inline app/models/relationship.rb]: indent

File: app/models/user.rb [hunk 8]
```diff
@@ -0,0 +1,65 @@
+class User < ActiveRecord::Base
+	before_save :downcase_email
```
Comment (chuanhtuan) [inline app/models/user.rb]: indent

File: app/models/word.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+class Word < ActiveRecord::Base
+	has_one :answer
```
Comment (chuanhtuan) [inline app/models/word.rb]: indent

File: app/views/layouts/application.html.erb [hunk 10]
```diff
@@ -1,14 +1,20 @@
...
+  <title>Fels25</title>
   <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
   <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
   <%= csrf_meta_tags %>
 </head>
 <body>
-
-<%= yield %>
-
+		<%= render 'layouts/header' %>
```
Comment (chuanhtuan) [inline app/views/layouts/application.html.erb]: indent

File: app/views/layouts/application.html.erb [hunk 11]
```diff
@@ -1,14 +1,20 @@
...
-<%= yield %>
-
+		<%= render 'layouts/header' %> 
+		<div class=container>
+			<% flash.each do | message_type , message | %>
+				 <%= content_tag(:div, message, class: "alert alert-#{message_type}")%>
+			<% end %>
+			<%= yield %>
+		</div>
+		<%= render 'layouts/footer' %>
```
Comment (chuanhtuan) [inline app/views/layouts/application.html.erb]: use " instead of '

File: app/controllers/static_pages_controller.rb (hunk 1 above)
Comment (TheTrung) [inline app/controllers/static_pages_controller.rb]: removed

File: app/controllers/static_pages_controller.rb (hunk 2 above)
Comment (TheTrung) [inline app/controllers/static_pages_controller.rb]: moved all to index action

File: app/helpers/static_pages_helper.rb (hunk 3 above)
Comment (TheTrung) [inline app/helpers/static_pages_helper.rb]: converted > space.

File: app/models/activity.rb (hunk 4 above)
Comment (TheTrung) [inline app/models/activity.rb]: removed

File: app/models/lesson.rb (hunk 5 above)
Comment (TheTrung) [inline app/models/lesson.rb]: converted > 2 spaces

File: app/models/question.rb (hunk 6 above)
Comment (TheTrung) [inline app/models/question.rb]: converted > 2 spaces

File: app/models/relationship.rb (hunk 7 above)
Comment (TheTrung) [inline app/models/relationship.rb]: fixed

File: app/models/user.rb (hunk 8 above)
Comment (TheTrung) [inline app/models/user.rb]: fixed

File: app/models/word.rb (hunk 9 above)
Comment (TheTrung) [inline app/models/word.rb]: fixed

File: app/helpers/static_pages_helper.rb [hunk 12]
```diff
@@ -0,0 +1,5 @@
+module StaticPagesHelper
+  def logged_in?
```
Comment (chuanhtuan) [inline app/helpers/static_pages_helper.rb]: SessionHelper

File: app/controllers/static_pages_controller.rb [hunk 13]
```diff
@@ -0,0 +1,12 @@
+class StaticPagesController < ApplicationController
+  def index
+    case params[:p]
+    when "categories"
```
Comment (chuanhtuan) [inline app/controllers/static_pages_controller.rb]: remove

File: app/models/user.rb [hunk 14]
```diff
@@ -0,0 +1,65 @@
+class User < ActiveRecord::Base
+  before_save :downcase_email
+
+  has_many :activities, dependent: :destroy
+
+  has_many :active_relationships, class_name: "Relationship",
+                                  foreign_key: "follower_id",
+                                  dependent: :destroy
+
+    has_many :passive_relationships, class_name: "Relationship",
```
Comment (chuanhtuan) [inline app/models/user.rb]: indent

File: app/views/static_pages/home.html.erb [hunk 15]
```diff
@@ -0,0 +1,29 @@
+<% if logged_in? %>
+  <div class="row">
+    <aside class="col-md-4">
+      <section class="user_info">
+        <%= render 'shared/user_info' %>
```
Comment (chuanhtuan) [inline app/views/static_pages/home.html.erb]: "

File: config/routes.rb [hunk 16]
```diff
@@ -1,56 +1,9 @@
 Rails.application.routes.draw do
-  # The priority is based upon order of creation: first created -> highest priority.
-  # See how all your routes lay out with "rake routes".
 
-  # You can have the root of your site routed with "root"
-  # root 'welcome#index'
+  root 'static_pages#index'
```
Comment (chuanhtuan) [inline config/routes.rb]: root 'static_pages#home'

File: app/helpers/static_pages_helper.rb (hunk 12 above)
Comment (TheTrung) [inline app/helpers/static_pages_helper.rb]: removed

File: app/controllers/static_pages_controller.rb (hunk 13 above)
Comment (TheTrung) [inline app/controllers/static_pages_controller.rb]: removed

File: app/models/user.rb (hunk 14 above)
Comment (TheTrung) [inline app/models/user.rb]: ok

File: app/views/static_pages/home.html.erb (hunk 15 above)
Comment (TheTrung) [inline app/views/static_pages/home.html.erb]: replaced

File: config/routes.rb (hunk 16 above)
Comment (TheTrung) [inline config/routes.rb]: fixed

**Pull request merged**